src/lightning/fabric/utilities/logger.py

```diff
@@ -129,10 +130,12 @@ def _sanitize_params(params: Dict[str, Any]) -> Dict[str, Any]:
 
     """
     for k in params:
-        # convert relevant np scalars to python types first (instead of str)
-        if isinstance(params[k], (np.bool_, np.integer, np.floating)):
-            params[k] = params[k].item()
-        elif type(params[k]) not in [bool, int, float, str, Tensor]:
+        if _NUMPY_AVAILABLE:
+            import numpy as np
+
+            if isinstance(params[k], (np.bool_, np.integer, np.floating)):
+                params[k] = params[k].item()
+        if type(params[k]) not in [bool, int, float, str, Tensor]:
```

Comment:
This can stay an elif:
```suggestion
        elif type(params[k]) not in [bool, int, float, str, Tensor]:
```